numpy/core/src/multiarray/iterators.c

```diff
@@ -1262,10 +1262,14 @@ PyArray_MultiIterFromObjects(PyObject **mps, int n, int nadd, ...)
     int i, ntot, err=0;
 
     ntot = n + nadd;
-    if (ntot < 1 || ntot > NPY_MAXARGS) {
+    if (ntot < 0) {
```

Comment:
How does one hit the code path where `ntot < 0`? If there's a concise / practical way to do that, coverage shows this is never true so a test would be useful.

Conversely, if this can't be hit by any practical code path by calling NumPy public API, is the guard really needed?

Reply:
I guess the guard is needed because of C API exposure.